assert/assertions.go

```diff
@@ -429,14 +429,27 @@ func samePointers(first, second interface{}) bool {
 // to a type conversion in the Go grammar.
 func formatUnequalValues(expected, actual interface{}) (e string, a string) {
 	if reflect.TypeOf(expected) != reflect.TypeOf(actual) {
-		return fmt.Sprintf("%T(%#v)", expected, expected),
-			fmt.Sprintf("%T(%#v)", actual, actual)
+		return fmt.Sprintf("%T(%s)", expected, truncatingFormat(expected)),
+			fmt.Sprintf("%T(%s)", actual, truncatingFormat(actual))
 	}
 	switch expected.(type) {
 	case time.Duration:
 		return fmt.Sprintf("%v", expected), fmt.Sprintf("%v", actual)
 	}
-	return fmt.Sprintf("%#v", expected), fmt.Sprintf("%#v", actual)
+	return truncatingFormat(expected), truncatingFormat(actual)
+}
+
+// The output of the error message is eventually read a line at time by a bufio.Scanner
```

Comment:
I think the function is clean, simple and easy to understand in terms of what it does. Do you think this comment is therefore redundant?

If you feel the need to have a comment here, would it not make more sense to follow the Go docstring format (i.e. starting with the function name) and describe 'why' instead of 'how'?

Reply:
I clean it up a little.  I do think having a little comment on why is important since I don't think it's obvious to everyone that the go testing framework has a limit on line length for it's error messages.

Reply:
It's a good point. I guess to clarify a bit: IMO it's perfectly fine to have a comment stating 'why' about the code but it's a bit redundant if the code is clean & simple and the comment states 'how'. e.g.
```
// ParseUserInput gets the user input and parses it to ourSpec struct
func ParseUserInput(userInput string) ourSpec {
...
}
```
is redundant IMO, whereas:
```
// ParseUserInput guards against bad input and converts to our standard data struct
func ParseUserInput(userInput string) ourSpec {
...
}
```